hazelcast/src/main/java/com/hazelcast/instance/GroupProperty.java

```diff
@@ -548,6 +548,18 @@
      */
     QUERY_OPTIMIZER_TYPE("hazelcast.query.optimizer.type", QueryOptimizerFactory.Type.RULES.toString()),
 
+
+    /**
+     * Cache de-serialized entries. Decreases number of de-serializations needed to evaluate a query when true.
+     * It allows to trade some memory for CPU cycles.
+     *
+     * Default: true
```

Comment:
We have also `MapConfig#optimizeQueries` which has a `false` default value. Keeping `QUERY_CACHE_ENTRIES` sync with that makes sense to me. So my vote is setting it to `false`.

Otherwise we may have 4 or 5 copy of the same value at the same time on a node : originalValue + backupValue+ cachedValue + queryCachedValue + nearCachedValue.